tools/docker/i2gw/Dockerfile

```diff
@@ -0,0 +1,9 @@
+# Use distroless as minimal base image to package the manager binary
```

Comment:
I don't think we want to maintain a Docker image for this, mind pulling this out?

Reply:
He @robscott, I added some comments above the makefile. This is more for developing, building and pushing image easily for contributors/users if they needed. The makefile for image supports multi-arch image build and push, pretty useful when users want to develop i2gw and try to host it with image privately.